go/vt/mysqlctl/mysql_daemon.go

```diff
@@ -101,12 +101,32 @@ func (fmd *FakeMysqlDaemon) SlaveStatus() (*proto.ReplicationStatus, error) {
 	return fmd.CurrentSlaveStatus, nil
 }
 
+// ExcludeTables copies the current schema, excluding tables specified by name.
+func ExcludeTables(sd *proto.SchemaDefinition, excludeTables []string) *proto.SchemaDefinition {
+	copy := *sd
+	filteredTables := make(proto.TableDefinitions, 0, len(sd.TableDefinitions))
+	for _, table := range sd.TableDefinitions {
+		excluded := false
+		for _, excludedTableName := range excludeTables {
+			if table.Name == excludedTableName {
```

Comment:
This, unfortunately, doesn't match what excludeTables is supposed to do. excludeTables []string is a string of regexes that are filtered, not just a list of exact table names that get filtered.

Take a look here: https://github.com/youtube/vitess/blob/master/go/vt/mysqlctl/schema.go#L45

Actually, now that I look at the code more carefully, it seems like we actually do all the filtering out on the vitess-side of things, not in MySQL. I think you should be able to pull the filtering code out into its own function, and just re-use it in both the real and the fake.